client/packages/core/src/instaql.js

```diff
@@ -333,43 +317,29 @@ function isBefore(startCursor, direction, [e, a, _v, t]) {
 }
 
 function runDataloadAndReturnObjects(store, etype, direction, pageInfo, dq) {
-  const startCursor = pageInfo?.["start-cursor"];
-  const toRemove = [];
-  const res = datalogQuery(store, dq)
-    .sort((tripleA, tripleB) => {
-      const tsA = tripleA[3];
-      const tsB = tripleB[3];
+  const aid = idAttr(store, etype).id;
+  const idVecs = datalogQuery(store, dq)
+    .sort(([_, tsA], [__, tsB]) => {
       return direction === "desc" ? tsB - tsA : tsA - tsB;
-    })
-    .reduce((res, triple) => {
-      const [e, a, v] = triple;
-      if (shouldIgnoreAttr(store.attrs, a)) {
-        return res;
-      }
-      const attr = store.attrs[a];
-      const [_, attrEtype, label] = attr["forward-identity"];
-      if (attrEtype !== etype) {
-        return res;
-      }
-
-      if (
-        startCursor &&
-        a === startCursor[1] &&
-        isBefore(startCursor, direction, triple)
-      ) {
-        toRemove.push(e);
-      }
-
-      res[e] = res[e] || {};
-      res[e][label] = v;
-      return res;
-    }, {});
+    });
 
-  // remove anything before our start cursor
-  for (const e of toRemove) {
-    delete res[e];
+  let objects = {}
+  const startCursor = pageInfo?.["start-cursor"];
+  const blobAttrs = s.blobAttrs(store, etype);
+  for (const [id, time] of idVecs) {
+    if (
+      startCursor &&
+      aid === startCursor[1] &&
+      isBefore(startCursor, direction, [id, aid, id, time])
```

Comment:
We'll have to rewrite this a bit if we add sorting on user-defined attributes.

Reply:
🫡